lib/Common/Memory/Recycler.cpp

```diff
@@ -5362,6 +5436,49 @@ Recycler::BackgroundMark()
     }
 }
 
+void
+Recycler::BackgroundMarkWeakRefs()
+{
+#if ENABLE_WEAK_REFERENCE_REGIONS
+    auto iterator = this->weakReferenceRegionList.GetIterator();
+    while (iterator.Next())
+    {
+        RecyclerWeakReferenceRegion region = iterator.Data();
+        RecyclerWeakReferenceRegionItem<void*> *items = region.GetPtr();
+        size_t count = region.GetCount();
+        for (size_t index = 0; index < count; ++index)
+        {
+            RecyclerWeakReferenceRegionItem<void*> &item = items[index];
+            if (item.ptr == nullptr)
+            {
+                continue;
+            }
+
+            if (((uintptr_t)item.ptr & 0x1) == 0x1)
+            {
+                // This weak reference is already marked
+                continue;
+            }
+
+            if (item.heapBlock == nullptr)
+            {
+                item.heapBlock = this->FindHeapBlock(item.ptr);
+                if (item.heapBlock == nullptr)
+                {
+                    // This isn't a real weak reference, ignore it
```

Comment:
I was somewhat concerned about if someone wanted to have a region of `Var`s, and happened to put some tagged values in there. Also, in another branch I tried to convert some other dictionaries to use these regions, and those dictionaries sometimes store references to statically allocated values, which are of course not recycler managed. These cases don't seem like errors, and the correct behavior is to keep the references around since they aren't being collected.